plugins/content/bower/index.js

```diff
@@ -803,93 +803,112 @@ function addPackage (plugin, packageInfo, options, cb) {
         return addCb(err);
       }
 
-      // don't duplicate component.name, component.version
-      db.retrieve(plugin.type, { name: package.name, version: package.version }, function (err, results) {
+      const targetAttribute = pkgMeta.targetAttribute;
+
+      async.some([ 'componenttype', 'extensiontype', 'menutype', 'themetype' ], (type, asyncCallback) => {
+        if (!targetAttribute) return asyncCallback();
+
+        db.retrieve(type, { targetAttribute: targetAttribute }, (err, results) => {
+          asyncCallback(err, results && results.length);
+        });
+      }, (err, targetAttributeExists) => {
         if (err) {
           logger.log('error', err);
           return addCb(err);
         }
 
-        if (results && 0 !== results.length) {
-          // don't add duplicate
-          if (options.strict) {
-            return addCb(new PluginPackageError("Can't add plugin: plugin already exists!"));
-          }
-          return addCb(null);
+        if (targetAttributeExists) {
+          return addCb(new PluginPackageError(`Can't add plugin: targetAttribute already exists!`));
```

Comment:
Please could this go in the language file? I've just made a PR which should mean that this gets shown on the plugin upload error modal (if it doesn't already)

Reply:
👍 shall wait for that PR to be merged so we don't encounter conflicts.

Reply:
Other PR is merged now :)